commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceShipmentItemClayTable.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.order.web.internal.frontend;
+
+import com.liferay.commerce.frontend.ClayTable;
+import com.liferay.commerce.frontend.ClayTableSchema;
+import com.liferay.commerce.frontend.ClayTableSchemaBuilder;
+import com.liferay.commerce.frontend.ClayTableSchemaBuilderFactory;
+import com.liferay.commerce.frontend.CommerceDataSetDataProvider;
+import com.liferay.commerce.frontend.Filter;
+import com.liferay.commerce.frontend.Pagination;
+import com.liferay.commerce.model.CommerceOrderItem;
+import com.liferay.commerce.model.CommerceShipmentItem;
+import com.liferay.commerce.order.web.internal.model.ShipmentItem;
+import com.liferay.commerce.service.CommerceOrderItemService;
+import com.liferay.commerce.service.CommerceShipmentItemService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.util.ParamUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+@Component(
+	immediate = true,
+	property = {
+		"commerce.data.provider.key=" + CommerceShipmentItemClayTable.NAME,
+		"commerce.table.name=" + CommerceShipmentItemClayTable.NAME
+	},
+	service = {ClayTable.class, CommerceDataSetDataProvider.class}
+)
+public class CommerceShipmentItemClayTable
+	implements ClayTable, CommerceDataSetDataProvider<ShipmentItem> {
+
+	public static final String NAME = "commerceShipmentItems";
+
+	@Override
+	public int countItems(HttpServletRequest httpServletRequest, Filter filter)
+		throws PortalException {
+
+		long commerceShipmentId = ParamUtil.getLong(
+			httpServletRequest, "commerceShipmentId");
+
+		return _commerceShipmentItemService.getCommerceShipmentItemsCount(
+			commerceShipmentId);
+	}
+
+	@Override
+	public ClayTableSchema getClayTableSchema() {
+		ClayTableSchemaBuilder clayTableSchemaBuilder =
+			_clayTableSchemaBuilderFactory.clayTableSchemaBuilder();
+
+		clayTableSchemaBuilder.addField("sku", "sku");
+		clayTableSchemaBuilder.addField("orderId", "order-id");
+		clayTableSchemaBuilder.addField("ordered", "ordered");
+		clayTableSchemaBuilder.addField("shipped", "shipped");
+		clayTableSchemaBuilder.addField("toSend", "to-send");
```

Comment:
Order matters here, so there should be line breaks between each of these statements.